examples/cms-starter/src/data.ts

```diff
@@ -0,0 +1,293 @@
+            ...collectionReferences,
+            {
+                id: collection.id,
+                name: collection.name,
+            },
+        ])
+    }
+}
+
+export function computeFieldConfig(existingFields: ManagedCollectionField[], dataSource: DataSource) {
+    const result: FieldConfig[] = []
+    const fields = dataSource.fields
+
+    for (const [name, field] of Object.entries(fields)) {
+        const fieldId = generateHash(name)
+        let newField: ManagedCollectionField | null = null
+
+        const existingField = existingFields.find(field => field.id === fieldId)
+        if (existingField) {
+            newField = existingField
+        } else if (field.type === "reference") {
+            newField = {
+                id: fieldId,
+                name,
+                type: field.multiple ? "multiCollectionReference" : "collectionReference",
+                collectionId: COLLECTIONS_SYNC_MAP.get(field.reference)?.[0].id ?? "",
+                userEditable: false,
+            }
+        } else {
+            const fieldType = FIELD_MAPPING[field.type][0] ?? "string"
+            newField = {
+                id: fieldId,
+                name,
+                type: fieldType,
+                userEditable: false,
+            } as ManagedCollectionField
+        }
+
+        let reference: ReferenceField | null = null
+        if (newField && field.type === "reference") {
+            if (newField.type === "string") {
+                reference = {
+                    type: field.multiple ? "multiCollectionReference" : "collectionReference",
+                    source: field.reference,
+                    destination: COLLECTIONS_SYNC_MAP.get(field.reference)?.[0].id ?? null,
+                }
+            } else if (newField.type === "collectionReference" || newField.type === "multiCollectionReference") {
+                reference = {
+                    type: newField.type,
+                    source: field.reference,
+                    destination: newField.collectionId || null,
+                }
+            }
+
+            assert(
+                true,
+                "Expected reference field to be mapped to a collection reference or multi collection reference"
+            )
+        }
+
+        if (field.type === "enum") {
+            assert(newField?.type === "enum", "Expected enum field to be mapped to an enum field")
+            newField.cases = field.options.map(option => ({
+                id: option,
+                name: option,
+            }))
+        }
+
+        result.push({
+            source: {
+                name,
+                type: field.type,
+                ignored: !existingField,
+            },
+            field: newField,
+            reference,
+        })
+    }
+
+    return result
+}
+
+const CELL_BOOLEAN_VALUES = ["Y", "yes", "true", "TRUE", "Yes", 1, true]
+
+export function getFieldValue(field: FieldConfig, value: unknown) {
```

Comment:
I think we don't need to demonstrate this kind of parsing logic, because that is just specific to the plugin. We should just read the values directly and have them be valid in the source data. Can add a comment to say that in a real world you'll want to have more complex logic.